geode-core/build.gradle

```diff
@@ -367,10 +350,11 @@ dependencies {
   integrationTestCompile(project(':geode-dunit')) {
     exclude module: 'geode-core'
   }
+  integrationTestImplementation(project(':geode-log4j')) {
```

Comment:
I think it would be better if geode-core tests did not depend on geode-log4j. If we always test geode-core with geode-log4j, we will never if there are parts of the system which will just fail without geode-log4j.

Could we instead just add plain old log4j-core here as a test dependency instead, or whatever we expect users to use?

Reply:
Too many of the integration tests are end-to-end server or locator requiring logging and/or alerting.

Reply:
Note: I'm still very committed to having geode-log4j in the classpath for integrationTests which do spin up servers/locators, perform logging and alerting, etc. These tests are not debuggable without logs. We would also have to find a new home for all integrationTests which have validation involving logging and alerting, and that's not something I want to tackle in this PR.

Reply:
Would adding log4j itself as a dependency not solve the issue of getting logs in the tests? It would just mean that geode-core would be tested without whatever special functionality geode-log4j adds.

I'm still concerned that we need to test geode-core *without* geode-log4j, if we are going to claim that geode-log4j is optional. 

I think being forced to move tests of alerting, etc. to geode-log4j would be valuable, because it would show us what functionality doesn't work without geode-log4j.